### PR TITLE
fix: can retrive COFT with COT automatically

### PR DESCRIPTION
**Pull request by WillieRuemmele**
### What does this PR do?
fixes retrieval of COFT

### What issues does this PR fix or reference?

https://github.com/forcedotcom/cli/issues/1233, https://github.com/forcedotcom/cli/issues/1241, https://github.com/forcedotcom/cli/issues/1262, @W-10222972@

### Functionality Before
`retrieve -m CustomFieldTranslation` without a CustomObjectTranslation in the org would only pull the CustomFieldTranslation files, which were invalid by themselves without a CustomObjectTranslation parent file

### Functionality After
a retrieve will retrieve both COFT/COT and write an empty COT if one doesn't exist in the org

### CODE REVIEW

File: src/convert/transformers/decomposedMetadataTransformer.ts
Comment (WillieRuemmele) [inline src/convert/transformers/decomposedMetadataTransformer.ts:89]: this could use the `isAddressable` field, because COFT is the only metadata type to set either of those. It just seemed like the naming and intention could become confusing if I used `isAddressable` 🤷‍♂️

File: src/convert/transformers/decomposedMetadataTransformer.ts
Comment (mshanemc) [inline src/convert/transformers/decomposedMetadataTransformer.ts:65]: +1

File: src/registry/types.ts
Comment (mshanemc) [inline src/registry/types.ts:28]: add the jsdoc for that the option does, like all the others (if we decide to use the new option).  Otherwise, modify isAddressable's explanation

File: src/convert/transformers/decomposedMetadataTransformer.ts
Comment (mshanemc) [reply src/convert/transformers/decomposedMetadataTransformer.ts:89]: I'm gonna defer to Steve's opinion on this one.

My question is whether it's valid to have `isAddressable:false` without `requiresParent:true`, or vice versa?  

Or would you say that what we really meant was `unaddressableWithoutParent` ?

**Review by mshanemc — APPROVED** [review]
I'd ship it as is if it fixed the problem.  I'd like to see @shetzel opine on whether this should be a new property on `MetadataType` or not.

If it should be a new prop AND it's only valid in combination with isAddressable, then I'd want a validation to be added for that in registryValidation (verify that no childTypes exist with an invalid combination of those properties AND that they can only be used on childTypes).

File: src/convert/transformers/decomposedMetadataTransformer.ts
Comment (shetzel) [reply src/convert/transformers/decomposedMetadataTransformer.ts:89]: I know there are more entities that are unaddressable, and that field is useful and makes sense so that one should remain.  I agree with Shane that `unaddressableWithoutParent` is better.  COT and COFT are unfortunate anomalies that hopefully will never happen again.  `unaddressableWithoutParent` is the nicest description of this disastrous pair.

**Review by shetzel — APPROVED** [review]

**Pull request merged**
